x-pack/legacy/plugins/monitoring/public/components/setup_mode/check_internal_monitoring.js

```diff
@@ -0,0 +1,123 @@
+import { npSetup } from 'ui/new_platform';
+import { ELASTIC_WEBSITE_URL, DOC_LINK_VERSION } from 'ui/documentation_links';
+
+const { cloud } = npSetup.plugins;
+const isCloudEnabled = Boolean(cloud && cloud.isCloudEnabled);
+let optOut = false;
+
+const getElement = () => document.getElementById('internal-monitoring-warning');
+
+export const destroyInternalMonitoringWarning = () => {
+  const el = getElement();
+  el && unmountComponentAtNode(el);
+};
+
+const callOutProps = {
+  title: i18n.translate('xpack.monitoring.internalCollectionDetectedTitle', {
+    defaultMessage: 'Internal Collection Detected',
+  }),
+  color: 'warning',
+};
+
+const description = i18n.translate('xpack.monitoring.internalMonitoringDescription', {
+  defaultMessage: `It appears you are using "Internal Collection" for Stack Monitoring. 
+    This method of monitoring will no longer be supported in the next major release (8.0.0). 
+    Please follow "Enter setup mode" steps and migrate to external monitoring with Metricbeat.`,
+});
+
+const InternalMonitoringDetectedBanner = props => (
+  <EuiCallOut {...callOutProps}>
+    <EuiFlexGroup gutterSize="s">
+      <EuiFlexItem grow={true}>
+        <p>
+          {description}{' '}
+          <EuiLink
+            target="_blank"
+            href={`${ELASTIC_WEBSITE_URL}guide/en/beats/metricbeat/${DOC_LINK_VERSION}/monitoring-metricbeat-collection.html`}
+          >
+            Show more
+          </EuiLink>
+        </p>
+      </EuiFlexItem>
+      <EuiFlexItem
+        grow={true}
+        style={{ marginTop: -28, maxWidth: 30, position: 'absolute', right: 16 }}
+      >
+        <EuiButtonIcon
+          aria-label="Opt-out of internal monitoring warning"
+          color="danger"
+          onClick={() => (optOut = true) && destroyInternalMonitoringWarning()}
+          iconType="cross"
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+    <EuiButton
+      onClick={() => props.toggleSetupMode(true)}
+      iconType="flag"
+      size="s"
+      iconSide="right"
+    >
+      {i18n.translate('xpack.monitoring.setupMode.enter', {
+        defaultMessage: 'Enter setup mode',
+      })}
+    </EuiButton>
+  </EuiCallOut>
+);
+
+const mountComponent = toggleSetupMode => {
+  const props = { toggleSetupMode };
+  const el = getElement();
+  el && render(<InternalMonitoringDetectedBanner {...props} />, el);
+};
+
+export const checkInternalMonitoring = async ($injector, toggleSetupMode) => {
+  if (isCloudEnabled || optOut) {
+    return;
+  }
+
+  // Intentionally deferring to allow initial API calls to get through first
+  await setTimeout(async () => {
+    const { cluster_uuid: clusterUUID } = $injector.get('globalState');
+    if (!clusterUUID) {
+      return;
+    }
+
+    const { getPageData } = await import('../../lib/get_page_data');
+    getPageData(
+      $injector,
+      `../api/monitoring/v1/clusters/${clusterUUID}/elasticsearch/indices?show_system_indices=true`
```

Comment:
Hmm. I'm not sure we need to do it this way. What if we send down a flag from the server indicating if we should show the callout or not?

Reply:
I was thinking about that, but decided not to tamper with the API since all of this will be removed in `8.0` (from what I understand). So, I think it's easier (and maybe more safer) to utilize existing APIs.

Up to you though, I think adding a flag should be fairly straight forward and can actually have more coverage, since it would be more ideal to have an equivalent of `GET _cat/indices` instead

Reply:
Yea this just seems inefficient. Fetching all indices and then doing a client-side check if a single `-mb-` exists.

Let's try a server side solution for this. We should be able to do a simple, efficient query to determine if any `-mb-` indices and just return down a boolean from the server.

Reply:
It would clean a lot of this up 💯 But, how do you feel a separate route instead of a flag? I don't really want to fuse "temporary" logic with the current index API. And, it'll be more primitive since I don't need any aggregation and can just do a simple `_cat/indices` with 
`-mb-` check wdyt?

Reply:
Well, I think we need to determine _which_ pages this should appear and that might influence our decision here.

Are we thinking _all_ pages, or just select pages?